component/src/stake/component.rs

```diff
@@ -229,7 +231,9 @@ pub(crate) trait StakingImpl: StateWriteExt {
                 self.set_validator_bonding_state(
                     identity_key,
                     Unbonding {
-                        unbonding_epoch: self.current_unbonding_end_epoch().await?,
+                        unbonding_epoch: self
+                            .current_unbonding_end_epoch_for(&identity_key)
```

Comment:
`current_unbonding_end_epoch` is now `current_unbonding_end_epoch_for(validator)` , so that it's not possible to use a generic unbonding period rather than the validator-specific one (which may be shorter, if the validator is already partly through unbonding).